### ISSUE TITLE
Unexpected errors do not error

### ISSUE THREAD

**Issue by OliverJAsh** [4Catalyzer/eslint-plugin-ts-expect#13]
Hi, thanks for your work on this project!

If my tests have an _unexpected_ error, e.g.

```ts
const add = (a: number, b: number): number => a + b;

add(1);
```

… then ESLint will return no errors.

The TSLint variant inside dtslint did not have this problem.

**Comment by jquense** [comment]
Hey thanks for trying this out. The plugin here isn't meant as a drop in replacement for dtslint as a tool. In particular, _eslint_ doesn't do general type checking for typescript, linting is a layer on top of that. The expectation is you'd still do your normal typechecking with the typescript compiler. Lint rules are sugar on top of that. 

In fact tho this what dtslint does, it runs `tsc --noEmit` against your code as well as tslint. This project is a bit more scoped, providing only the special test-specific lint rule to read and check specific comments.

**Comment by OliverJAsh** [comment]
> The expectation is you'd still do your normal typechecking with the typescript compiler

But if we do that, `tsc` will fail due to _expected_ errors, which we don't want it to do. 🤔 We only want it to fail due to unexpected errors.

**Comment by jquense** [comment]
fair point. There isn't much we can do tho as an eslint plugin, You'd need a typescript plugin perhaps? Our workflow for tests is that we don't explicitly typecheck them, beyond normal editor support. For test related things that we want to test/make assertions on, we use the `$Expect` comments. Beyond that tho, running `tsc` on test code doesn't offer much from a _testing_ perspective (in terms of failing tests via CI, it obviously has DX value).

**Comment by OliverJAsh** [comment]
> running tsc on test code doesn't offer much from a testing perspective

My worry is that if we don't type check the tests, then when the source code changes, there might be new (unexpected) type errors in the tests. In that case, the tests wouldn't fail during CI, so no-one would notice unless they opened the tests in their editor!

**Comment by OliverJAsh** [comment]
Would it not be possible for the lint rule to throw a linting error if the file contains errors which are not marked as expected? This is what the TSLint rule appears to be doing.

**Comment by OliverJAsh** [comment]
FWIW this version works as I would expect: https://github.com/ibezkrovnyi/eslint-plugin-expect-type
